File: solutions/zig/bob/1/bob.zig
pub fn response(s: []const u8) []const u8 {
    if (s.len == 0)
        return "Fine. Be that way!";

    var question = false;
    var shouting = true;
    var text = false;
    var silence = true;
    for (s) |c| {
        switch (c) {
            'a'...'z' => {
                shouting = false;
                text = true;
                silence = false;
                question = false;
            },
            'A'...'Z' => {
                text = true;
                silence = false;
                question = false;
            },
            ' ', '\t', '\n', '\r' => continue,
            '?' => question = true,
            else => {
                silence = false;
                question = false;
            },
        }
    }

    if (silence)
        return "Fine. Be that way!";
    if (question and shouting and text)
        return "Calm down, I know what I'm doing!";
    if (question)
        return "Sure.";
    if (shouting and text)
        return "Whoa, chill out!";

    return "Whatever.";
}

File: solutions/go/bob/1/bob.go
// Package bob should have a package comment that summarizes what it's about.
package bob

import "strings"

// Hey should have a comment documenting it.
func Hey(remark string) string {
	remark = strings.TrimSpace(remark)
	question := false
	if strings.HasSuffix(remark, "?") {
		question = true
	}

	yell := false
	if strings.ToUpper(remark) == remark && strings.ToLower(remark) != remark {
		yell = true
	}

	if remark == "" {
		return "Fine. Be that way!"
	} else if question && yell {
		return "Calm down, I know what I'm doing!"
	} else if question {
		return "Sure."
	} else if yell {
		return "Whoa, chill out!"
	} else {
		return "Whatever."
	}
}
